src/Microsoft.Identity.Client/Instance/AadInstanceDiscovery.cs

```diff
@@ -47,7 +49,10 @@ public bool TryGetValue(string host, out InstanceDiscoveryMetadataEntry instance
             Uri authority,
             RequestContext requestContext)
         {
-            if (!TryGetValue(authority.Host, out var entry))
+            bool foundInCache = TryGetValue(authority.Host, out var entry);
+            TraceWrapper.WriteLine("GetMetadataEntryAsync - response from cache? " + foundInCache);
```

Comment:
Trace does not exist in .NetStandard 1.3 and in UAP, so I added a simple wrapper that does nothing on these platforms (we can get tracing in via NuGet packages, but probably overkill)

Reply:
The extensions libraries all use Trace.  I really wonder why we have a logging callback instead of just hooking up a trace provider.  Or at least use ILogger from microsoft.extensions.logging.

Reply:
Probably to not take a dependency on another logging library ...

Reply:
Something we should absolutely consider more. 

---
In reply to: [285189965](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1157#discussion_r285189965) [](ancestors = 285189965)